presto-main/src/main/java/io/prestosql/execution/SqlQueryExecution.java

```diff
@@ -414,7 +415,7 @@ private PlanRoot doAnalyzeQuery()
 
         // plan query
         PlanNodeIdAllocator idAllocator = new PlanNodeIdAllocator();
-        LogicalPlanner logicalPlanner = new LogicalPlanner(stateMachine.getSession(), planOptimizers, idAllocator, metadata, sqlParser, statsCalculator, costCalculator, stateMachine.getWarningCollector());
+        LogicalPlanner logicalPlanner = new LogicalPlanner(stateMachine.getSession(), planOptimizers, idAllocator, metadata, new TypeAnalyzer(sqlParser, metadata), statsCalculator, costCalculator, stateMachine.getWarningCollector());
```

Comment:
should it based to `SqlQueryExecution` constructor?

Reply:
I doesn't seem necessary in this case. SqlQueryExecution has all it needs to create it and also needs the parser, so no need to further pollute the constructor.